rust/xaynet-analytics/src/database/analytics_event/adapter.rs

```diff
@@ -31,33 +31,34 @@ impl AnalyticsEventAdapter {
             name: name.into(),
             event_type,
             timestamp,
-            screen_route_field,
+            screen_route_field: screen_route_field.map(|field| field.into()),
         }
     }
 }
 
 impl<'event> IsarAdapter<'event> for AnalyticsEventAdapter {
+    fn get_oid(&self) -> String {
+        format!("{}-{}", self.name, self.timestamp)
+    }
+
     fn into_field_properties() -> IntoIter<FieldProperty> {
         // NOTE: properties need to be ordered by type. Properties with the same type need to be ordered alphabetically
         // https://github.com/isar/isar-core/blob/1ea9f27edfd6e3708daa47ac6a17995b628f31a6/src/schema/collection_schema.rs
         vec![
```

Comment:
i think this is not necessary anymore with the upgrade to the newer Isar version. I'll check it during the first round of reviews and if so i'll remove it